src/app/containers/PageHandlers/withData.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { shape } from 'prop-types';
+import articlePropTypes from '../../models/propTypes/article';
+import ErrorMain from '../ErrorMain';
+
+const WithData = Component => {
+  return function WithDataComponent({ data }) {
+    if (data) {
+      const { data: articleData, status } = data;
+      try {
+        return status === 200 && articleData ? (
+          <Component articleData={articleData} />
+        ) : (
+          <ErrorMain status={status} />
+        );
+      } catch {
+        return <ErrorMain status={status} />;
+      }
+    }
+
+    return <ErrorMain status={status} />;
```

Comment:
These Error mains can likely be consolidated to a single call, so if data is bad, or status != 200 or the try catch is thrown, it falls to the error. 

Also I think status is undefined here, but I haven't run it

Reply:
ahh good spot on status being undefined, although I think the function requires an return statement at the end, so we either return `null` or simply return the error component with a status code of 500. 

Thoughts?

Reply:
Confirmed, assume data is `null` the final return statement is needed otherwise nothing renders.

Reply:
I think you could do this, cant see any obvious issues, but ive not run it 🙈 😂 
```
const WithData = Component => ({ data }) => {
  const { data: articleData, status } = data;

  if (status === 200 && articleData) {
    try {
      return <Component articleData={articleData} />
    } catch (err) {
      logger.error('Oh noos' + err);
    }
  }

  return <ErrorMain status={status || 500} />;
};
```

Reply:
If the data prop was null or anything other than an object this would fail with `Cannot read property 'data' of null`.

Reply:
Although we change the default prop for data to be an empty object?

Reply:
Ermm whoops, yep I was wrong, data should be required and have no default, updated code